nncf/tensorflow/pruning/filter_pruning/algorithm.py

```diff
@@ -240,6 +240,12 @@ def set_pruning_level(self, pruning_level: float,
         if run_batchnorm_adaptation:
             self._run_batchnorm_adaptation()
 
+    @property
+    def maximal_compression_rate(self) -> float:
+        if self.prune_flops:
+            return 1 - self._min_prunable_flops / max(self.full_flops, 1)
+        return 1.0
```

Comment:
This is not correct statement. The pruning algorithm can not prune all model parameters.

Reply:
Yes. But at the same time currently it is possible to set any (channel) pruning rate and it will not throw an exception to it. I think this logic needs to be implemented first in order to support the behavior of the property value you suggest.